### ISSUE TITLE
protocol can not worked with <audio>

### ISSUE THREAD

**Issue by jareguo** [atom/electron#1123]
We had a problem that protocols can not used in audio elements.
It can be reproduced by `<audio src="library://background.mp3" autoplay="true"></audio>`.

Here is the demo: https://github.com/jareguo/test-atom-audio-protocol

@jwu

**Comment by jareguo** [comment]
and this is how we register protocol
```js
  Protocol.registerProtocol('library', function(request) {
      var url = decodeURIComponent(request.url);
      var data = Url.parse(url);
      var relativePath = data.hostname;
      if ( data.pathname ) {
          relativePath = Path.join( relativePath, data.pathname );
      }
      var file = Path.join('library', relativePath);
      return new Protocol.RequestFileJob(file);
  });
```

**Comment by paulcbetts** [comment]
@jareguo It seems like you're just trying to construct a file:// URL - why not just use them directly?

**Comment by jwu** [comment]
Sure we can, but we hide the file-system path in RenderContext side, which means the web page developer didn't known where the audio file exists in the file-system/server but only the relative path.

**Comment by zcbenz** [comment]
I can confirm this, from the devtools the connection was `(canceled)`, which is odd.

**Comment by samdoiron** [comment]
+1 just ran into this, however it happened when I intercepted the `file://` protocol.

**Comment by deepak1556** [comment]
Was looking into this, tried registering a `file_protocol_handler` for some custom scheme during job_factory setup to check whether <audio> or <video> tag has some rule for custom schemes but things worked fine. So the problem is with adapter_request_job not performing range request properly here.

**Comment by zcbenz** [comment]
Fixed by #2474.
